users/test/test_views.py

```diff
@@ -3,20 +3,169 @@
 
 from django.contrib.auth import get_user_model
 from django.contrib.auth.tokens import PasswordResetTokenGenerator
+from django.core import mail
 from django.utils.encoding import smart_bytes
 from django.utils.http import urlsafe_base64_encode
 from faker import Faker
 from rest_framework import status
 from rest_framework.exceptions import PermissionDenied
 from rest_framework.reverse import reverse
-from rest_framework.test import APIClient, APIRequestFactory, APITestCase
+from rest_framework.test import APIClient, APITestCase
 
-from users.views import PasswordResetAPIView, PasswordResetEmailView
+from .mocks import test_user, test_user_2
 
 fake = Faker()
 User = get_user_model()
-password_reset_email_view = PasswordResetEmailView.as_view()
-password_reset_api_view = PasswordResetAPIView.as_view()
+
+
+class TestUserList(APITestCase):
+    user: Any
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        cls.user = User.objects.create_user(**test_user)
+
+    @property
+    def bearer_token(self) -> str:
+        login_url = reverse("login")
+        response = self.client.post(
+            login_url,
+            data={
+                "email": test_user["email"],
+                "password": test_user["password"],
+            },
+            format="json",
+        )
+        return response.data.get("access")  # type: ignore[no-any-return,attr-defined]
+
+    def test_create_user(self) -> None:
+        url = reverse("users")
+        data = {
+            "username": fake.user_name(),
+            "email": fake.email(),
+            "password": fake.password(),
+        }
+        outbox = len(mail.outbox)
+        response = self.client.post(url, data=data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(len(mail.outbox), outbox + 1)
+
+    def test_user_login(self) -> None:
+        url = reverse("login")
+        response = self.client.post(
+            url,
+            data={
+                "email": test_user["email"],
+                "password": test_user["password"],
+            },
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_user_list(self) -> None:
+
+        url = reverse("users")
+        self.client.credentials(  # type: ignore[attr-defined]
```

Comment:
is the type ignore necessary here?

Reply:
Yes it is.
for the `error: "Client" has no attribute "credentials"  [attr-defined]` error